### PR TITLE
First commit for project skeleton

### PR DESCRIPTION
**Pull request by AzmiTouil**

### CODE REVIEW

File: gamification-twitter-services/pom.xml [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    <dependency>
+      <groupId>org.exoplatform.commons</groupId>
+      <artifactId>commons-component-common</artifactId>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.exoplatform.commons</groupId>
+      <artifactId>commons-comet-service</artifactId>
+      <scope>provided</scope>
+    </dependency>
```
Comment (boubaker) [inline gamification-twitter-services/pom.xml]: Useless imports

```suggestion
```

File: gamification-twitter-services/pom.xml [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    <dependency>
+      <groupId>org.exoplatform.social</groupId>
+      <artifactId>social-component-core</artifactId>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.exoplatform.social</groupId>
+      <artifactId>social-component-service</artifactId>
+      <scope>provided</scope>
+    </dependency>
```
Comment (boubaker) [inline gamification-twitter-services/pom.xml]: Useless import
```suggestion
```

File: gamification-twitter-services/pom.xml [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+      <groupId>com.fasterxml.jackson.core</groupId>
+      <artifactId>jackson-annotations</artifactId>
+      <version>${org.jackson.version}</version>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.projectlombok</groupId>
+      <artifactId>lombok</artifactId>
+      <scope>provided</scope>
+    </dependency>
```
Comment (boubaker) [inline gamification-twitter-services/pom.xml]: Useless import
```suggestion
```

File: gamification-twitter-services/pom.xml [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+    <dependency>
+      <groupId>org.hsqldb</groupId>
+      <artifactId>hsqldb</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>com.github.scribejava</groupId>
+      <artifactId>scribejava-apis</artifactId>
+    </dependency>
```
Comment (boubaker) [inline gamification-twitter-services/pom.xml]: Useless import
```suggestion
```

File: gamification-twitter-services/pom.xml [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+            <artifactId>aspectjtools</artifactId>
+            <version>1.9.1</version>
+          </dependency>
+          <dependency>
+            <groupId>org.aspectj</groupId>
+            <artifactId>aspectjweaver</artifactId>
+            <version>1.9.1</version>
+          </dependency>
+        </dependencies>
+      </plugin>
```
Comment (boubaker) [inline gamification-twitter-services/pom.xml]: Workaround not needed anymore

```suggestion
      <plugin>
        <groupId>com.jcabi</groupId>
        <artifactId>jcabi-maven-plugin</artifactId>
      </plugin>
```

File: gamification-twitter-services/pom.xml [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+      <!-- we need to overide the enforcer rules here because lombok doesn't match it -->
+      <plugin>
+        <artifactId>maven-enforcer-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>check-java-compat</id>
+            <phase>none</phase>
+          </execution>
+        </executions>
+      </plugin>
```
Comment (boubaker) [inline gamification-twitter-services/pom.xml]:
```suggestion
```

File: gamification-twitter-webapp/src/main/webapp/WEB-INF/web.xml [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+		version="3.0"
+		metadata-complete="true"
+		xmlns="http://java.sun.com/xml/ns/javaee"
+		xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+		xsi:schemaLocation="http://java.sun.com/xml/ns/javaee http://java.sun.com/xml/ns/javaee/web-app_3_0.xsd">
+
+	<display-name>gamification-twitter</display-name>
+	<absolute-ordering />
+
+	<filter>
```
Comment (boubaker) [inline gamification-twitter-webapp/src/main/webapp/WEB-INF/web.xml]: Change tabs to spaces
```suggestion
    <filter>
```

File: gamification-twitter-webapp/src/main/webapp/WEB-INF/web.xml [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+		<filter-name>ResourceRequestFilter</filter-name>
+		<filter-class>org.exoplatform.portal.application.ResourceRequestFilter</filter-class>
+	</filter>
+
```
Comment (boubaker) [inline gamification-twitter-webapp/src/main/webapp/WEB-INF/web.xml]: Missing 

```xml

  <listener>
    <listener-class>org.exoplatform.container.web.PortalContainerConfigOwner</listener-class>
  </listener>

```

File: gamification-twitter-webapp/webpack.common.js [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+        './src/main/webapp/vue-app/**/*.js',
+        './src/main/webapp/vue-app/**/*.vue',
+      ],
+    }),
+    new VueLoaderPlugin()
+  ],
+  externals: {
+    vuetify: 'Vuetify',
+    vue: 'Vue',
+    jquery: '$'
```
Comment (boubaker) [inline gamification-twitter-webapp/webpack.common.js]: jquery deprecated to avoid using it

```suggestion
```

File: package-lock.json [hunk 10]
```diff
@@ -0,0 +1,6 @@
+{
+  "name": "gamification-twitter",
+  "lockfileVersion": 2,
+  "requires": true,
+  "packages": {}
+}
```
Comment (boubaker) [inline package-lock.json]: To delete, file not used

```suggestion
```

File: pom.xml [hunk 11]
```diff
@@ -0,0 +1,140 @@
...
+  <scm>
+    <connection>scm:git:git://github.com/meeds-io/gamification-twitter.git</connection>
+    <developerConnection>scm:git:ssh://git@github.com/meeds-io/gamification-twitter.git</developerConnection>
+    <url>https://github.com/meeds-io/gamification-twitter</url>
+    <tag>HEAD</tag>
+  </scm>
+  <properties>
+    <org.exoplatform.social.version>6.5.x-SNAPSHOT</org.exoplatform.social.version>
+    <org.exoplatform.platform-ui.version>6.5.x-SNAPSHOT</org.exoplatform.platform-ui.version>
+    <org.jackson.version>2.4.2</org.jackson.version>
```
Comment (boubaker) [inline pom.xml]:
```suggestion
```

File: pom.xml [hunk 12]
```diff
@@ -0,0 +1,140 @@
...
+        <version>${project.version}</version>
+        <type>zip</type>
+      </dependency>
+
+      <dependency>
+        <groupId>com.fasterxml.jackson.jaxrs</groupId>
+        <artifactId>jackson-jaxrs-base</artifactId>
+        <version>${org.jackson.version}</version>
+      </dependency>
+
```
Comment (boubaker) [inline pom.xml]: To move to `depmgt` project

```suggestion
```

File: gamification-twitter-webapp/src/main/webapp/META-INF/exo-conf/configuration.xml [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+    <set-method>registerChangePlugin</set-method>
+    <!-- The full qualified name of the PortalContainerDefinitionChangePlugin -->
+    <type>org.exoplatform.container.definition.PortalContainerDefinitionChangePlugin</type>
+    <priority>500</priority>
+    <init-params>
+      <value-param>
+        <name>apply.default</name>
+        <value>true</value>
+      </value-param>
+      <object-param>
```
Comment (boubaker) [inline gamification-twitter-webapp/src/main/webapp/META-INF/exo-conf/configuration.xml]: Missing profile declaration

```suggestion
      <object-param>
      <values-param>
          <name>add.profiles</name>
          <value>gamification-twitter</value>
        </values-param>
      <object-param>
```

Comment (AzmiTouil) [reply gamification-twitter-webapp/src/main/webapp/META-INF/exo-conf/configuration.xml]: Done!

**Review by boubaker — APPROVED** [review]

**Pull request merged**